lib/cose/key/okp.rb

```diff
@@ -14,9 +15,53 @@ def self.enforce_type(map)
         end
       end
 
+      def self.from_pkey(pkey)
+        curve = Curve.by_pkey_name(pkey.oid) || raise("Unsupported edwards curve #{pkey.oid}")
+        attributes = { crv: curve.id }
+
+        begin
+          asymmetric_key = pkey.private_to_der
+          private_key = OpenSSL::ASN1.decode(asymmetric_key).value.last.value
+          curve_private_key = OpenSSL::ASN1.decode(private_key).value
+          attributes.merge!({ d: curve_private_key })
+        rescue OpenSSL::PKey::PKeyError
+          asymmetric_key = pkey.public_to_der
+          public_key_bit_string = OpenSSL::ASN1.decode(asymmetric_key).value.last.value
+          attributes.merge!({ x: public_key_bit_string })
```

Comment:
Yes, but rubber ducking here in public just in case (Cunningham's law 😉)

Taking Ed25519 test 2 vectors from https://www.rfc-editor.org/rfc/rfc8032#section-7 (just like the [OpenSSL gem test suite](https://github.com/ruby/openssl/blob/e0718e4d2c0f751719e136f3cb79616daa3f6686/test/openssl/test_pkey.rb#L79-L114)):
```ruby
priv_pem = <<~EOF
  -----BEGIN PRIVATE KEY-----
  MC4CAQAwBQYDK2VwBCIEIEzNCJso/5banbbDRuwRTg9bijGfNaumJNqM9u1PuKb7
  -----END PRIVATE KEY-----
EOF
priv = OpenSSL::PKey.read(priv_pem)
cose_priv = COSE::Key::OKP.from_pkey(priv)
public_key_hex = cose_priv.x.unpack1("H*")
puts public_key_hex.chars.each_slice(32).map(&:join).join("\n")
# 3d4017c3e843895a92b70aa74d1b7ebc
# 9c982ccf2ec4968cc0cd55f12af4660c
```